sourcegraph/sourcegraph
# Code Insights: Move from index approach for handling series to series ids

Closes https://github.com/sourcegraph/sourcegraph/issues/29664

## Context 

Prior to this PR, we used series indexes for tracking the current series form for edit/delete operations. Turned out that the indexes handling isn't a reliable approach when we use as keys series id. For example, let's say we have two series forms. Then we delete the first one. Visually we have only one series form, but the index in the code hasn't been changed because the second form still has the same key prop (which is series id). 

## How to test

- Create two series 
- Remove the first one
- Try to edit any field in the series form
- Try to create another series form

Expected behavior: All series have their values.
